### PR TITLE
ci: proto up-to-date check

### PR DESCRIPTION
**Pull request by infiniteregrets**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,6 +64,17 @@ jobs:
         with:
           crate: cargo-sort
           version: "^1.0"
+      - name: check proto build up-to-date
+        run: |
+          cargo build --features prost-build          
+          if git diff proto/; then
```
Comment (vrongmeal) [inline .github/workflows/ci.yml:70]: Why do we want to check if `proto/` changed or not?

Comment (vrongmeal) [reply .github/workflows/ci.yml:70]: Oh yeah, but the `else` branch is still:

```
git diff --exit-code src/api.rs
```

Doesn't having only this suffice?

Comment (infiniteregrets) [reply .github/workflows/ci.yml:70]: yeah.. gotta test it out, i just wrote it lol

Comment (infiniteregrets) [reply .github/workflows/ci.yml:70]: no i meant like, you open a PR to upgrade protos and lets say you forgot to regen src/api.rs but git diff with main is ok, means all checks pass. i wanted to consider that but the bash code may not be enough/right

Comment (vrongmeal) [reply .github/workflows/ci.yml:70]: Yup, the git diff for `api.rs` should be enough for everything.

Comment (vrongmeal) [reply .github/workflows/ci.yml:70]:
```
git diff --exit-code src/api.rs
```

Won't be `0` if any updates are uncommitted

Comment (infiniteregrets) [reply .github/workflows/ci.yml:70]: ah ok! thanks for reminding me of that

**Pull request merged**